### PR TITLE
Add function for Harbour socket API and disabled debug code HB_CDX_DBGCODE

### PR DESCRIPTION
**Pull request by lailton**
* include/hbsocket.h
  * src/harbour.def
  * src/rtl/hbsocket.c
  * src/rtl/hbsockhb.c
    + added functions for Harbour socket API
      HB_SOCKETSETNOSIGPIPE( hSocket, lValue ) --> lSuccess

  * src/rdd/dbfcdx/dbfcdx1.c
    ! disabled debug code HB_CDX_DBGCODE.

### CODE REVIEW

File: src/rtl/hbsocket.c [hunk 1]
```diff
@@ -2848,6 +2860,21 @@ int hb_socketSetNoDelay( HB_SOCKET sd, HB_BOOL fNoDelay )
    return ret;
 }
 
+int hb_socketSetNoSigPipe( HB_SOCKET sd, HB_BOOL fNoSigPipe )
+{
+#if defined( SO_NOSIGPIPE )
```
Comment (tuffnatty) [inline src/rtl/hbsocket.c:2865]: isn't this condition always true due to #ifndef...#define above?

File: src/rtl/hbsocket.c [hunk 2]
```diff
@@ -2848,6 +2860,21 @@ int hb_socketSetNoDelay( HB_SOCKET sd, HB_BOOL fNoDelay )
...
+   int val = fNoSigPipe ? 1 : 0, ret;
+   ret = setsockopt( sd, SOL_SOCKET, SO_NOSIGPIPE, ( const char * ) &val, sizeof( val ) );
+   hb_socketSetOsError( ret != -1 ? 0 : HB_SOCK_GETERROR() );
```
Comment (tuffnatty) [inline src/rtl/hbsocket.c:2868]: I'd parenthsize the condition to ease human suffering

File: src/rdd/dbfcdx/dbfcdx1.c [hunk 3]
```diff
@@ -54,8 +54,8 @@
 #  define HB_CDX_PACKTRAIL
 #endif
 
-#define HB_CDX_DBGCODE
 /*
+#define HB_CDX_DBGCODE
```
Comment (tuffnatty) [inline src/rdd/dbfcdx/dbfcdx1.c:58]: Why this change? It seems to be totally unrelated to the socket part. IMHO needs a separate PR and some explanation

**Review by tuffnatty — CHANGES_REQUESTED** [review]

File: src/rtl/hbsocket.c (hunk 1 above)
Comment (lailton) [reply src/rtl/hbsocket.c:2865]: No, if not defined HB_SOCKET_OFF then the SO_NOSIGPIPE is not defined too.

File: src/rtl/hbsocket.c (hunk 2 above)
Comment (lailton) [reply src/rtl/hbsocket.c:2868]: I have followed the other procedure to be the same way.

File: src/rdd/dbfcdx/dbfcdx1.c (hunk 3 above)
Comment (lailton) [reply src/rdd/dbfcdx/dbfcdx1.c:58]: It is a debug define for RARE situation with problem to CDX, if you check the line on prg dbfcdx1.c contains other defined already commented.
